src/interactions/brush.d.ts

```diff
@@ -0,0 +1,41 @@
+import type {Rendered} from "../transforms/basic.js";
+
+/** Options for the brush transform. */
+type BrushOptions = {
+  /**
+   * The brush’s selection mode determines the contents of the plot’s value
+   * property when the user manipulates the brush:
+   * * **data** - default; the selected data
+   * * **extent** - the selection extent, in data space
+   *
+   * The extent is an object with properties *x*: [x1, x2] for brushX, *y*: [y1,
+   * y2] for brushY, and both *x* and *y* for brush. Additionally, when
+   * faceting, it contains the facet’s *fx* and *fy* properties.
+   */
+  selectionMode?: "data" | "extent";
```

Comment:
I don’t think we’ll want this option; we should always provide both. To do this, the brush’s value could be the selection array decorated with the extent. I.e., _selection_ = […the subset of _data_ that is selected by the brush…], and _selection_.x = [_x1_, _x2_], _selection_.y = [_y1_, _y2_]. Or perhaps better, _selection_.x1, _selection_.x2, _selection_.y1, _selection_.y2, more like we do to represent the extent of bins with the bin transform. This way you always get both representations.